File: GreenDaoTest/app/src/main/java/com/huskar/greendaotest/greendaotest/MainActivity.java
package com.huskar.greendaotest.greendaotest;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import org.greenrobot.greendao.query.Query;

import java.util.Date;

public class MainActivity extends AppCompatActivity {
    private NoteDao noteDao;
    private Query<Note> notesQuery;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // get the note DAO
        DaoSession daoSession = ((App) getApplication()).getDaoSession();
        noteDao = daoSession.getNoteDao();

        notesQuery = noteDao.queryBuilder().orderAsc(NoteDao.Properties.Text).build();

    }

    private void insert(){
        Note note = new Note();
        note.setText("aaa");
        note.setComment("aaa");
        note.setDate(new Date());
        note.setType(NoteType.TEXT);
        noteDao.insert(note);
    }
}

File: KotlinDemo/app/src/main/java/com/huskar/kotlin/kotlindemo/MainActivity.kt
package com.huskar.kotlin.kotlindemo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        my_textview.text = "hello kotlin"
        my_textview.setOnClickListener {
            Toast.makeText(this,"hello kotlin",Toast.LENGTH_LONG).show()
            b();
        }
    }
    fun MainActivity.b() {
        this.my_textview.setText("hello fun b")
    }
}
